velodyne_pointcloud/include/velodyne_pointcloud/datacontainerbase.h

```diff
@@ -141,53 +178,79 @@ class DataContainerBase
     config_.fixed_frame = fixed_frame;
     config_.target_frame = target_frame;
 
-    config_.transform = fixed_frame.compare(target_frame) != 0;
-    if (config_.transform && !tf_ptr)
-    {
-      tf_ptr = boost::shared_ptr<tf::TransformListener>(new tf::TransformListener);
-    }
```

Comment:
Hmm this should be here if we don't like to transform the cloud. It has been overlooked here https://github.com/ros-drivers/velodyne/pull/344#discussion_r452368123 too.

Reply:
@spuetz See https://github.com/ros-drivers/velodyne/pull/351/files#diff-8617d0d04cbc690d4e7529f8a263d278R144 and the conditions there for allocating the `tf_buffer` and `tf_listener`.